lib/lolitado/db.rb

```diff
@@ -0,0 +1,98 @@
+module Lolitado
+  class Database
+
+    attr_accessor :db
+
+    def initialize db
+      @db = db
+    end
+    
+    
+    def query sql
+      result = db[sql].all
+      return result
+    end
+
+    def multiple_query sql
+      splited_sql = sql.split(';')
+      splited_sql.each do |each_sql|
+        query(each_sql)
+      end
+    end
+    
+    def query_duration sql, type = true, waiting_time = 10
+      start = Time.now
+      if type
+        result = query(sql, waiting_time)
+      else
+        result = query_empty(sql, waiting_time)
+      end
+      finish = Time.now
+      msecs = (finish - start) * 1000.0
+      hash = {'result' => result, 'duration' => msecs}
+      return hash
+    end
+
+    ## Expect to return data for the query.
+    ## If the result is empty then it will be called again until waiting_time reach.
+    def query_wait sql, waiting_time = 10
+      result = db[sql].all
+      if result.length == 0
```

Comment:
I'm not sure the class of "result" has to be Array or Hash,  if so, use "result.empty?" or "result.length.zero?" maybe  more ruby.

Reply:
The result have to be Array.
For yes, result.empty? is fine, it could be more ruby.